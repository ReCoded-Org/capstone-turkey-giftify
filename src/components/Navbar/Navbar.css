.navbar {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: baseline;
  align-content: stretch;
  font-family: "Roboto", sans-serif;
  padding-top: 1%;
  background-color: white;
}

.navbarLogo {
  background-color: yellow;
  margin: 0.5% 0% 0% 1.5%;
}

.navbarLinks {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: baseline;
  align-content: stretch;
}

.eachLinksInNavbar {
  padding: 0 15px;
  color: #3f3b3b;
  text-decoration: none;
}

.eachLinksInNavbar:hover {
  color: black;
  text-decoration: none;
  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  transition: 0.3s;
}

#dropdownBasicButton {
  color: #3f3b3b;
  background-color: rgba(255, 255, 255, 0);
  border-color: rgba(255, 255, 255, 0);
}

.btn {
  margin-right: 5px;
}

.navbarProfile {
  margin: 0% 1% 0% 1.5%;
}

@media only screen and (min-width: 120px) and (max-width: 991px) {
  .navbarMobile {
    background-color: white;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .menuBars {
    font-size: 2rem;
    background: none;
    margin: 0 0 3.5% 3.5%;
  }

  .menuBars > svg {
    color: black;
  }

  .navMenu {
    background-color: white;
    width: 250px;
    height: 100vh;
    display: flex;
    justify-content: center;
    position: fixed;
    top: 0;
    left: -100%;
    transition: 850ms;
  }

  .navMenu.active {
    left: 0;
    transition: 350ms;
  }

  .navText {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 8px 0px 8px 16px;
    list-style: none;
    height: 60px;
    list-style-type: none;
    color: black;
    text-decoration: none;
  }

  .navText a {
    text-decoration: none;
    color: #f5f5f5;
    font-size: 18px;
    width: 95%;
    height: 100%;
    display: flex;
    align-items: center;
    padding: 0 16px;
    border-radius: 4px;
  }

  .navText a:hover {
    color: black;
    text-decoration: none;
    text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    transition: 0.3s;
  }

  .navMenuItems {
    width: 100%;
  }

  .navbarToggle {
    background-color: white;
    width: 100%;
    height: 80px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }

  .navLi {
    list-style-type: none;
  }

  .navLiLang {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-items: center;
    align-content: stretch;
    margin-left: 7%;
  }

  #dropdownBasicButton {
    color: black;
  }

  .navLiLoginSign {
    margin-left: 7%;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-items: center;
    align-content: stretch;
  }

  .navbarMobileLogo {
    margin-right: 5%;
  }
}

@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");