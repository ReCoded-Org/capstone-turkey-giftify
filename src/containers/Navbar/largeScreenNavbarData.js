export const largeScreenNavbarData = [
  {
    id: "a",
    target: "/",
    cName: "eachLinksInNavbar",
    name: "Home",
  },
  {
    id: "b",
    target: { pathname: "/", hash: "#search" },
    cName: "eachLinksInNavbar",
    name: "Search",
  },
  {
    id: "c",
    target: "/requests",
    cName: "eachLinksInNavbar",
    name: "Requests",
  },
  {
    id: "d",
    target: "/donations",
    cName: "eachLinksInNavbar",
    name: "Donations",
  },
  {
    id: "e",
    target: "/about",
    cName: "eachLinksInNavbar",
    name: "About",
  },
  {
    id: "f",
    target: "/contactUs",
    cName: "eachLinksInNavbar",
    name: "Contact Us",
  },
];
