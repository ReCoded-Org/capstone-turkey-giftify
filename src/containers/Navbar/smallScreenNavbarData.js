import React from "react";
import {
  HouseDoor,
  Search,
  Gift,
  Heart,
  InfoCircle,
  Envelope,
} from "react-bootstrap-icons";

export const smallScreenNavbarData = [
  {
    cName: "navLi",
    target: "/",
    cNameLink: "navText",
    name: "Home",
    component: <HouseDoor />,
  },
  {
    cName: "navLi",
    target: { pathname: "/", hash: "#search" },
    cNameLink: "navText",
    name: "Search",
    component: <Search />,
  },
  {
    cName: "navLi",
    target: "/requests",
    cNameLink: "navText",
    name: "Requests",
    component: <Gift />,
  },
  {
    cName: "navLi",
    target: "/donations",
    cNameLink: "navText",
    name: "Donations",
    component: <Heart />,
  },
  {
    cName: "navLi",
    target: "/about",
    cNameLink: "navText",
    name: "About",
    component: <InfoCircle />,
  },
  {
    cName: "navLi",
    target: "/contactUs",
    cNameLink: "navText",
    name: "Contact Us",
    component: <Envelope />,
  },
];
