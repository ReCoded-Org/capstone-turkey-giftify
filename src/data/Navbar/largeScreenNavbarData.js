export const largeScreenNavbarData = [
  {
    id: "a",
    target: "/",
    name: "Home",
  },
  {
    id: "b",
    target: "/search",
    name: "Search",
  },
  {
    id: "c",
    target: "/requests",
    name: "Requests",
  },
  {
    id: "d",
    target: "/donations",
    name: "Donations",
  },
  {
    id: "e",
    target: "/about",
    name: "About",
  },
  {
    id: "f",
    target: "/contactUs",
    name: "Contact Us",
  },
];
