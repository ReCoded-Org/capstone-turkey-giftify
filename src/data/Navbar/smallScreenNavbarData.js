import React from "react";
import {
  HouseDoor,
  Search,
  Gift,
  Heart,
  InfoCircle,
  Envelope,
} from "react-bootstrap-icons";

export const smallScreenNavbarData = [
  {
    id: "a",
    target: "/",
    name: "Home",
    component: <HouseDoor />,
  },
  {
    id: "b",
    target: { pathname: "/", hash: "#search" },
    name: "Search",
    component: <Search />,
  },
  {
    id: "c",
    target: "/requests",
    name: "Requests",
    component: <Gift />,
  },
  {
    id: "d",
    target: "/donations",
    name: "Donations",
    component: <Heart />,
  },
  {
    id: "e",
    target: "/about",
    name: "About",
    component: <InfoCircle />,
  },
  {
    id: "f",
    target: "/contactUs",
    name: "Contact Us",
    component: <Envelope />,
  },
];
