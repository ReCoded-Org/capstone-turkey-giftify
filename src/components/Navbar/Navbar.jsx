import React, { useState, useEffect, useRef } from "react";
import "./Navbar.css";
import "bootstrap/dist/css/bootstrap.min.css";
import { Button, Dropdown, DropdownButton } from "react-bootstrap";
import {
  List,
  HouseDoor,
  Search,
  Gift,
  Heart,
  InfoCircle,
  Envelope,
  GeoAlt,
} from "react-bootstrap-icons";
import { Link } from "react-router-dom";

const NavbarForLargeScreens = () => {
  return (
    <div id="navbar">
      <div id="navbarLogo">
        <Link to="/">LOGO</Link>
      </div>
      <div id="navbarLinks">
        <Link to="/" className="eachLinksInNavbar">
          Home
        </Link>
        <Link
          to={{ pathname: "/", hash: "#search" }}
          className="eachLinksInNavbar"
        >
          Search
        </Link>
        <Link to="/requests" className="eachLinksInNavbar">
          Requests
        </Link>
        <Link to="/donations" className="eachLinksInNavbar">
          Donations
        </Link>
        <Link to="/about" className="eachLinksInNavbar">
          About
        </Link>
        <Link to="/contactUs" className="eachLinksInNavbar">
          Contact Us
        </Link>
        <DropdownButton id="dropdown-basic-button" title="Language">
          <Dropdown.Item href="#/action-1">TUR</Dropdown.Item>
          <Dropdown.Item href="#/action-2">ENG</Dropdown.Item>
        </DropdownButton>
      </div>
      <div id="navbarProfile">
        <Link to="/logIn">
          <Button
            variant="primary"
            style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
          >
            Log In
          </Button>
        </Link>
        <Link to="/signUp">
          <Button
            variant="primary"
            style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
          >
            Sign Up
          </Button>
        </Link>
      </div>
    </div>
  );
};

const NavbarForSmallScreen = () => {
  const [sidebar, setSidebar] = useState(false);

  const showSidebar = () => setSidebar(!sidebar);

  const boxRef = useRef(null);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (boxRef && boxRef.current.contains(event.target)) {
        setSidebar(true);
      } else {
        setSidebar(false);
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, boxRef);

  return (
    <>
      <div className="navbar-mobile">
        <Link to="#" className="menu-bars">
          <List onClick={showSidebar} />
        </Link>
        <Link to="/" className="navbar-mobile-logo">
          LOGO
        </Link>
      </div>
      <nav ref={boxRef} className={sidebar ? "nav-menu active" : "nav-menu"}>
        <ul className="nav-menu-items">
          <li className="navbar-toggle" onClick={showSidebar}>
            <Link to="#" className="menu-bars">
              <List />
            </Link>
          </li>

          <li className="nav-li nav-li-login-sign" onClick={showSidebar}>
            <Link to="/logIn">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Log In
              </Button>
            </Link>
          </li>
          <li className="nav-li nav-li-login-sign" onClick={showSidebar}>
            <Link to="/signUp">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Sign Up
              </Button>
            </Link>
          </li>

          <li className="nav-li" onClick={showSidebar}>
            <Link to="/" className="nav-text">
              <HouseDoor /> &nbsp; Home
            </Link>
          </li>
          <li className="nav-li" onClick={showSidebar}>
            <Link to={{ pathname: "/", hash: "#search" }} className="nav-text">
              <Search /> &nbsp; Search
            </Link>
          </li>
          <li className="nav-li" onClick={showSidebar}>
            <Link to="/requests" className="nav-text">
              <Gift /> &nbsp; Requests
            </Link>
          </li>
          <li className="nav-li" onClick={showSidebar}>
            <Link to="/donations" className="nav-text">
              <Heart /> &nbsp; Donations
            </Link>
          </li>
          <li className="nav-li" onClick={showSidebar}>
            <Link to="/about" className="nav-text">
              <InfoCircle /> &nbsp; About
            </Link>
          </li>
          <li className="nav-li" onClick={showSidebar}>
            <Link to="/contactUs" className="nav-text">
              <Envelope /> &nbsp; Contact Us
            </Link>
          </li>
          <li className="nav-li nav-li-lang">
            <GeoAlt style={{ color: "black" }} />
            <DropdownButton id="dropdown-basic-button" title="Language">
              <Dropdown.Item href="#/action-1" onClick={showSidebar}>
                TUR
              </Dropdown.Item>
              <Dropdown.Item href="#/action-2" onClick={showSidebar}>
                ENG
              </Dropdown.Item>
            </DropdownButton>
          </li>
        </ul>
      </nav>
    </>
  );
};

const getWindowDimensions = () => {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
};

const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  );

  useEffect(() => {
    const handleResize = () => {
      setWindowDimensions(getWindowDimensions());
    };

    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
  }, []);

  return windowDimensions;
};

const Navbar = () => {
  const { width } = useWindowDimensions();

  if (width < 950) {
    return <NavbarForSmallScreen />;
  } else {
    return <NavbarForLargeScreens />;
  }
};

export default Navbar;
