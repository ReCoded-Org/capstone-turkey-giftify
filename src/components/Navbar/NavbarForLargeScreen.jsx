import React from "react";
import { Link } from "react-router-dom";
import { Button } from "react-bootstrap";
import { useSelector, useDispatch } from "react-redux";
import firebase from "firebase/app";
import "firebase/auth";
import { largeScreenNavbarData } from "../../data/Navbar/largeScreenNavbarData.js";
import navBarLogo from "../../images/navBarLogo.png";

const NavbarForLargeScreen = () => {
  const { userId, userName } = useSelector((state) => state.user);
  const displayProfile = userId === "" ? "none" : "inline-block";
  const displayLogin = userId === "" ? "inline-block" : "none";
  const dispatch = useDispatch();

  const LogOut = () => {
    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch({
          type: "user/logout",
        });
        // Sign-out successful.
      })
      .catch((error) => {
        alert(error);
        // An error happened.
      });
  };

  return (
    <div className="navbar">
      <div className="navbarLogo">
        <Link to="/">
          <Image src={navBarLogo} />
        </Link>
      </div>
      <div className="navbarLinks">
        {largeScreenNavbarData.map((info) => (
          <Link to={info.target} className="eachLinksInNavbar" key={info.id}>
            {info.name}
          </Link>
        ))}
      </div>
      <div className="navbarProfile">
        <Link to="/logIn">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayLogin,
            }}
          >
            Login
          </Button>
        </Link>
        <Link to="/signUp">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayLogin,
            }}
          >
            Sign Up
          </Button>
        </Link>
        <Link to="/userProfile">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayProfile,
            }}
          >
            {userName}
          </Button>
        </Link>
        <Link to="/">
          <Button
            variant="primary"
            onClick={LogOut}
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayProfile,
            }}
          >
            Log Out
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default NavbarForLargeScreen;
