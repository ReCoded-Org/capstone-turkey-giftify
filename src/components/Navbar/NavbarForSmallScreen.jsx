import React, { useState, useEffect, useRef } from "react";
import { Link } from "react-router-dom";
import { List } from "react-bootstrap-icons";
import { Button } from "react-bootstrap";
import { smallScreenNavbarData } from "./../../data/Navbar/smallScreenNavbarData.js";

const NavbarForSmallScreen = () => {
  const [sidebar, setSidebar] = useState(false);

  const showSidebar = () => setSidebar(!sidebar);

  const boxRef = useRef(null);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (boxRef && boxRef.current.contains(event.target)) {
        setSidebar(true);
      } else {
        setSidebar(false);
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []);

  return (
    <>
      <div className="navbarMobile">
        <Link to="#" className="menuBars">
          <List onClick={showSidebar} />
        </Link>
        <Link to="/" className="navbarMobileLogo">
          LOGO
        </Link>
      </div>
      <nav ref={boxRef} className={sidebar ? "navMenu active" : "navMenu"}>
        <ul className="navMenuItems">
          <li className="navbarToggle" onClick={showSidebar}>
            <Link to="#" className="menuBars">
              <List />
            </Link>
          </li>

          <li className="navLi navLiLoginSign" onClick={showSidebar}>
            <Link to="/logIn">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Log In
              </Button>
            </Link>
          </li>
          <li className="navLi navLiLoginSign" onClick={showSidebar}>
            <Link to="/signUp">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Sign Up
              </Button>
            </Link>
          </li>
          {smallScreenNavbarData.map((info) => (
            <li className="navLi" onClick={showSidebar} key={info.id}>
              <Link to={info.target} className="navText">
                {info.component} &nbsp; {info.name}
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
};

export default NavbarForSmallScreen;
