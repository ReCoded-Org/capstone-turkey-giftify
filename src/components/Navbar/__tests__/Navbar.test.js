import React from "react";
import Navbar from "./../Navbar";
import renderer from "react-test-renderer";


it("Navbar rendered correctly", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
