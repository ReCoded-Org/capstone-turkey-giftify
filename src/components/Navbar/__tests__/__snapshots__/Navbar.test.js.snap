// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar"
>
  <div
    className="navbarLogo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      LOGO
    </a>
  </div>
  <div
    className="navbarLinks"
  >
    <a
      className="eachLinksInNavbar"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="eachLinksInNavbar"
      href="/#search"
      onClick={[Function]}
    >
      Search
    </a>
    <a
      className="eachLinksInNavbar"
      href="/requests"
      onClick={[Function]}
    >
      Requests
    </a>
    <a
      className="eachLinksInNavbar"
      href="/donations"
      onClick={[Function]}
    >
      Donations
    </a>
    <a
      className="eachLinksInNavbar"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="eachLinksInNavbar"
      href="/contactUs"
      onClick={[Function]}
    >
      Contact Us
    </a>
    <div
      className="dropdown"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-primary"
        disabled={false}
        id="dropdown-basic-button"
        onClick={[Function]}
        type="button"
      >
        Language
      </button>
    </div>
  </div>
  <div
    className="navbarProfile"
  >
    <a
      href="/logIn"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "borderRadius": "0.75rem",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        Log In
      </button>
    </a>
    <a
      href="/signUp"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "borderRadius": "0.75rem",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        Sign Up
      </button>
    </a>
  </div>
</div>
`;
